rules/__tests__/no-identical-title.test.js

```diff
@@ -81,6 +81,17 @@ ruleTester.run('no-identical-title', rule, {
       '    });',
       '});',
     ].join('\n'),
+    {
+      code: [
+        'describe("describe", () => {',
+        '    it(`testing ${someVar} with the same title`, () => {});',
+        '    it(`testing ${someVar} with the same title`, () => {});',
```

Comment:
I feel like this case could be statically analyzed and marked as invalid, but I'm also not super worried about digging into handling that edge case in this PR (the code would probably be a little complicated to validate this). Thoughts from others?

Reply:
What if I assign to/mutate `someVar` between test declarations?
Feels like a slippery slope to start with usage analysis. I think it's fine to leave as-is

Reply:
I agree that this would be nice to have, but for the time being it's fine as is.

Reply:
Definitely a slippery slope - didn't think about that case. 😅 Moving along... 😂